dev-docs/git-practices.md

```diff
@@ -0,0 +1,133 @@
+# Forks and PR workflows
+
+Changes to the Hail codebase are made through pull requests directly to the
+`main` branch of the `hail-is/hail` repository. Here are the steps to
+take when contributing for the first time and when making a pull request.
+
+
+## First time actions
+
+In order to keep the `hail-is/hail` repo clean, we ask that contributors develop
+features in branches on a personal fork. The first step then is to [fork
+the repository](https://github.com/hail-is/hail/fork).
+
+Once you've done so, clone the repository from your fork. This will set up a
+local copy of your fork with a single remote called `origin`. In this document,
+we will refer to `origin` as your developer fork, and `upstream` as `hail-is/hail`.
```

Comment:
Just a comment: I went with `hi` for my upstream. I use two-letter initials for all of y'all's repos.

Reply:
Ya shorter tends to be nicer, but I think in these kinds of documents I would prefer to use more standard language. Obviously if someone has a thorough enough understanding of git they'll replace it with whatever they want to use.